sui_core/src/authority/authority_store.rs

```diff
@@ -293,12 +296,25 @@ impl<const ALL_OBJ_VER: bool, S: Eq + Serialize + for<'de> Deserialize<'de>>
 
     /// Read an object and return it, or Err(ObjectNotFound) if the object was not found.
     pub fn get_object(&self, object_id: &ObjectID) -> Result<Option<Object>, SuiError> {
-        self.objects.get(object_id).map_err(|e| e.into())
+        let obj = self
```

Comment:
Should probably modify the RustDoc to say, read the latest version of an object

Reply:
This function now respects the tombstones in parent_sync, so its semantics are unchanged (as they have to be for existing tests to pass)